scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSchedulingPolicy.java

```diff
@@ -73,37 +114,78 @@ private static void initialize() {
         }
     }
 
-    @Override
-    public boolean isTaskExecutable(NodeSet selectedNodes, EligibleTaskDescriptor task) {
+    private static boolean isJobExecutableUsingLicense(JobDescriptorImpl job) {
+
+        logger.debug("Analysing job: " + ((JobDescriptorImpl) job).getInternal().getName());
+
+        // Retrieve required licenses names from the job generic informations
+        final String requiredLicenses = ((JobDescriptorImpl) job).getInternal()
+                                                                 .getRuntimeGenericInformation()
+                                                                 .get(REQUIRED_LICENSES);
+
+        // If it requires software licenses
+        if (requiredLicenses != null) {
+
+            initializeJobsLicenses();
+            logger.debug("Need to check licenses with " + properties.toString());
+
+            // Do not execute if one of the required license can not be obtained
+            final String[] requiredLicensesArray = requiredLicenses.split(",");
+            for (int i = 0; i < requiredLicensesArray.length; i++) {
 
-        logger.debug("Selected Nodes: " + selectedNodes);
+                final String currentRequiredLicense = requiredLicensesArray[i];
+                if (!canGetJobLicense(currentRequiredLicense)) {
+                    logger.debug("License for " + currentRequiredLicense + " not available, keep job pending");
+                    return false;
+                }
+            }
+            // Can be executed! Give it all required software licenses
+            for (int i = 0; i < requiredLicensesArray.length; i++) {
+                eligibleJobsDescriptorsLicenses.get(requiredLicensesArray[i]).add(job);
```

Comment:
it is very strange that you don't check the result of the add method.
Add is supposed to return true if the element can be added and false otherwise.